aim/web/ui/src/pages/Experiment/ExperimentStore.ts

```diff
@@ -17,13 +22,43 @@ function experimentEngine() {
     destroy: destroyExperiments,
   } = createResource<IExperimentData[]>(getExperiments);
 
+  function updateExperiment(name: string, description: string) {
+    const experimentData = experimentState.getState().data;
+    updateExperimentById(
+      { name, description, archived: experimentData?.archived },
+      experimentData?.id || '',
+    )
+      .then((res) => {
+        experimentState.setState((prev: any) => ({
+          ...prev,
```

Comment:
doesn't this work without `...prev`? zustand merges states instead of replacing

Reply:
You were right. I have removed it.